### DISCUSSION TITLE
Retain accessibility tags when loading PDF?

### DISCUSSION THREAD

**Discussion by ykolomatski** [Hopding/pdf-lib#1083] [Q&A]
Whenever I manipulate a PDF with pdf-lib and save the document, all the accessibility tags that were defined in the input PDF no longer exist.

Is there a way to instruct pdf-lib to retain them?

**Comment by pablodenadai** [comment]
Hi @ykolomatski did you find a solution to this issue? Or considered using another library?

**Reply by ykolomatski** [reply]
No, I didn't find a solution. No library or tool seem to support this functionality.

**Reply by pablodenadai** [reply]
Happy to work on it together if you have any leads on how to implement it :+1:

**Reply by pablodenadai** [reply]
Related: https://github.com/Hopding/pdf-lib/issues/895

**Reply by ykolomatski** [reply]
I am not familiar enough with the PDF standard to help implement that. 

Decompressing a tagged PDF with qpdf to analyze its contents is very useful to see the internal structure of the PDF. This, combined with the [PDF standard](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf), perhaps section 14 or 14.8, could be useful.

**Reply by ykolomatski** [reply]
"PDF explained" by John Whitington is also very useful as a reference to understand the structure of a PDF. It doesn't explain accessibility however.
